swank.asd

```diff
@@ -19,19 +19,87 @@
 ;; This code has been placed in the Public Domain.  All warranties
 ;; are disclaimed.
 
-(defpackage :swank-loader
-  (:use :cl))
+(defpackage #:swank-loader
+  (:use #:common-lisp #:asdf)
+  (:export #:init #:*source-directory*))
 
-(in-package :swank-loader)
+(in-package #:swank-loader)
 
-(defclass swank-loader-file (asdf:cl-source-file) ())
+(defun load-user-init-file ()
+  "Load the user init file, return NIL if it does not exist."
+  (load (merge-pathnames (user-homedir-pathname)
+                         (make-pathname :name ".swank" :type "lisp"))
+        :if-does-not-exist nil))
 
-;;;; after loading run init
+(defvar *source-directory*)
 
-(defmethod asdf:perform ((o asdf:load-op) (f swank-loader-file))
-  (load (asdf::component-pathname f))
-  (funcall (read-from-string "swank-loader::init") :reload t))
+(defclass no-load-file (cl-source-file) ())
 
-(asdf:defsystem :swank
-  :default-component-class swank-loader-file
-  :components ((:file "swank-loader")))
+;; (defmethod perform ((op load-op) (c no-load-file)) nil)
```

Comment:
If you don't define this method, it's actually loaded.

Maybe you want a secondary system that is compile-system'ed but not load-system'ed?

Reply:
Uncommented.